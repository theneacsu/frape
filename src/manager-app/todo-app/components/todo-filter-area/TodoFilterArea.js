import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
  setHideCompleted,
  setShowAll,
  setSearchTerm
} from '../../../../actions/todo-app/filters'
import styles from './TodoFilterArea.module.css'

export class TodoFilterArea extends Component {
  state = {
    showAll: true
  }

  onChangeCheckbox = e => {
    const checked = e.target.checked
    checked ? this.props.setHideCompleted() : this.props.setShowAll()
  }

  onChangeSearch = e => {
    const searchTerm = e.target.value
    this.props.setSearchTerm(searchTerm)
  }

  render() {
    return (
      <div className={styles.div}>
        <input
          className="input"
          type="text"
          placeholder="Search todos"
          onChange={this.onChangeSearch}
        />

        <input
          type="checkbox"
          className={styles.checkbox}
          id="completed-todos"
          onChange={this.onChangeCheckbox}
        />
        <label htmlFor="completed-todos">
          <span className={styles.span}>Hide completed</span>
        </label>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  setSearchTerm: searchTerm => dispatch(setSearchTerm(searchTerm)),
  setHideCompleted: () => dispatch(setHideCompleted()),
  setShowAll: () => dispatch(setShowAll())
})

export default connect(
  undefined,
  mapDispatchToProps
)(TodoFilterArea)
