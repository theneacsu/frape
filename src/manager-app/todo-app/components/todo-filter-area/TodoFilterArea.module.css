.div {
  display: flex;
  flex-direction: column;
}

.checkbox {
  margin-top: 1rem;
}

.checkbox {
  display: none;
}

.checkbox + label:before {
  content: '\2714';
  border: 0.1em solid #4cb69f;
  border-radius: 0.2em;
  display: inline-block;
  width: 1em;
  height: 1em;
  padding-left: 0.2em;
  padding-bottom: 0.3em;
  margin-right: 0.2em;
  vertical-align: bottom;
  color: transparent;
}

.checkbox + label:active:before {
  transform: scale(0);
}
.checkbox:checked + label:before {
  background-color: #4cb69f;
  border-color: #4cb69f;
  color: #fff;
}

.checkbox + label {
  display: block;
  margin: 2rem 2rem 0 0;
  cursor: pointer;
}

.checkbox {
  margin: 0 0 2rem 3rem;
}

.span {
  margin: 1rem;
}
