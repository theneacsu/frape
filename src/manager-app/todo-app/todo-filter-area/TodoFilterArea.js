import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
  setHideCompleted,
  setShowAll,
  setSearchTerm
} from '../../../actions/todo-app/filters'

export class TodoFilterArea extends Component {
  state = {
    showAll: true
  }

  onChangeCheckbox = e => {
    const checked = e.target.checked
    checked ? this.props.setHideCompleted() : this.props.setShowAll()
  }

  onChangeSearch = e => {
    const searchTerm = e.target.value
    this.props.setSearchTerm(searchTerm)
  }

  render() {
    return (
      <div className="todo-filters-box">
        <input
          className="todo-input"
          type="text"
          placeholder="Search todos"
          onChange={this.onChangeSearch}
        />

        <input
          type="checkbox"
          id="completed-todos"
          onChange={this.onChangeCheckbox}
        />
        <label htmlFor="completed-todos">
          <span>Hide completed</span>
        </label>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  setSearchTerm: searchTerm => dispatch(setSearchTerm(searchTerm)),
  setHideCompleted: () => dispatch(setHideCompleted()),
  setShowAll: () => dispatch(setShowAll())
})

export default connect(
  undefined,
  mapDispatchToProps
)(TodoFilterArea)
