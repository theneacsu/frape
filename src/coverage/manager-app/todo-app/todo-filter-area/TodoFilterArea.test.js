import React from 'react'
import { shallow } from 'enzyme'
import { TodoFilterArea } from '../../../../manager-app/todo-app/todo-filter-area/TodoFilterArea'

let wrapper, setSearchTermSpy, setHideCompletedSpy, setShowAllSpy

beforeEach(() => {
  setSearchTermSpy = jest.fn()
  setHideCompletedSpy = jest.fn()
  setShowAllSpy = jest.fn()

  wrapper = shallow(
    <TodoFilterArea
      setSearchTerm={setSearchTermSpy}
      setHideCompleted={setHideCompletedSpy}
      setShowAll={setShowAllSpy}
    />
  )
})

test('Should render correctly the TodoFilterArea component', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Should change the search term on input change', () => {
  const value = 'the'
  wrapper
    .find('input')
    .at(0)
    .simulate('change', {
      target: {
        value
      }
    })
  expect(setSearchTermSpy).toHaveBeenLastCalledWith(value)
})

test('Should change hideCompleted to true on checkbox change', () => {
  wrapper
    .find('input')
    .at(1)
    .simulate('change', {
      target: {
        checked: true
      }
    })
  expect(setHideCompletedSpy).toHaveBeenCalled()
})

test('Should change showAll to true on checkbox change', () => {
  wrapper
    .find('input')
    .at(1)
    .simulate('change', {
      target: {
        checked: false
      }
    })
  expect(setShowAllSpy).toHaveBeenCalled()
})
