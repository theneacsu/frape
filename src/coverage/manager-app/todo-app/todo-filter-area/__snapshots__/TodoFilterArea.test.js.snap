// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly the TodoFilterArea component 1`] = `
<div
  className="todo-filters-box"
>
  <input
    className="todo-input"
    onChange={[Function]}
    placeholder="search in todos"
    type="text"
  />
  <input
    id="completed-todos"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="completed-todos"
  >
    <span>
      Hide completed
    </span>
  </label>
</div>
`;
